torch/export/exported_program.py

```diff
@@ -412,12 +412,40 @@ def update_arg(old_arg, new_ph):
         new_placeholders = _get_placeholders(gm)
         new_outputs = list(gm.graph.nodes)[-1].args[0]
 
+        # To match the output target with correct input for input mutations
+        # need to find the correct new placeholder names with index
+        new_outputs_name = [output.name for output in new_outputs]
+        output_to_target_spec = {}
+        input_target_to_index = {
+            spec.arg.name: i
+            for i, spec in enumerate(self.graph_signature.input_specs)
+            if not isinstance(spec.arg, ConstantArgument)
+        }
+        output_target_to_index = {
+            spec.arg.name: i
+            for i, spec in enumerate(self.graph_signature.output_specs)
+            if not isinstance(spec.arg, ConstantArgument)
+        }
+        for (
+            output_node,
+            input_node,
+        ) in self.graph_signature.user_inputs_to_mutate.items():
+            input_index = input_target_to_index[input_node]
+            output_index = output_target_to_index[output_node]
+            output_to_target_spec[new_outputs_name[output_index]] = new_placeholders[
+                input_index
+            ].name
```

Comment:
I think this is correct? But it seems a little complicated..I wonder if we can just do:

```
        old_new_placeholder_map = dict(
                (spec.arg.name, new_placeholders[i].name) 
                for i, spec in enumerate(self.graph_signature.input_specs) 
                if not isinstance(spec.arg, ConstantArgument)
        )
        output_specs = [
            OutputSpec(
                spec.kind, 
                update_arg(spec.arg, new_outputs[i]), 
                old_new_placeholder_map.get(spec.target, spec.target)
            )
            for i, spec in enumerate(self.graph_signature.output_specs)
        ]
```

Reply:
That makes sense to me, since the target it an input spec name, we just need to map the new placeholder input name to the expected target spec.   Will update it.

Reply:
Updated it, please let me know. 🙏